webtau-test-server/src/main/java/org/testingisdocumenting/webtau/http/testserver/GraphQLResponseHandler.java

```diff
@@ -59,7 +59,7 @@ public GraphQLResponseHandler(GraphQLSchema schema, Handler additionalHandler) {
     @Override
     public void handle(String url, Request baseRequest, HttpServletRequest request,
                        HttpServletResponse response) throws IOException, ServletException {
-        if ("/graphql".equals(baseRequest.getOriginalURI())) {
+        if (baseRequest.getOriginalURI().startsWith("/graphql")) {
```

Comment:
Hmm, this doesn't play so nicely with the `GraphQLHttpConfiguration` concept because the URL may not be `/graphql`.  Perhaps we drop this check and rely purely on POST + validation of the request + response?

Reply:
it's tricky. The concept of `if (isGraphQLRequest) handleGraphQL() else if (useOtherHandler) doThat else 404` does make sense, especially since this is to support testing of both main handlers.
We could do some request introspection to see whether this is a valid graphql request. Basically most code that you already have in `handleGraphQLPathRequest` , just without auth check and handle call.

Reply:
thinking about this some more, is there anything that can tell us with near certainty that a request is a graphql request , outside of the endpoint? Only `query` is required, `operationName` and `variables` are optional.
Elastic search requests have a `query`, too, and possibly many custom endpoints.
Since this is a test server, it might be ok requiring the endpoint to at least start with `/graphql`, although it's admittedly not watertight anymore after this PR but hits 99% of our use cases.

Reply:
Sorry about the delay here.

In terms of detecting it's GraphQL I think looking for `query` and then trying to parse it as a GraphQL request is as good as you're going to get.  ES and others are highly unlikely to use GraphQL syntax in a `query` variable.

That said, given that hardly anyone will change from `/graphql` what you have is probably ok for the time being.  Might be worth calling this out in docs though.